sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/models/_models.py

```diff
@@ -936,5 +967,57 @@ def clear(self) -> None:
             self.ack_map.clear()
 
 
+class AckMapAsync:
+    """Async Ack map"""
+
+    def __init__(self) -> None:
+        self.ack_map: Dict[int, SendMessageErrorOptionsAsync] = {}
+
+    def add(self, ack_id: int, options: SendMessageErrorOptionsAsync) -> None:
+        """Add ack id to ack map
+
+        :param ack_id: The ack id. Required.
+        :type ack_id: int
+        :param options: The options. Required.
+        :type options: SendMessageErrorOptions
+        """
+        self.ack_map[ack_id] = options
+
+    def pop(self, ack_id: int) -> Optional[SendMessageErrorOptionsAsync]:
+        """Pop ack id from ack map
+
+        :param ack_id: The ack id. Required.
+        :type ack_id: int
+        :return: The options.
+        :rtype: SendMessageErrorOptions or None
+        """
+        return self.ack_map.pop(ack_id, None)
+
+    def get(self, ack_id: int) -> Optional[SendMessageErrorOptionsAsync]:
+        """Get ack id from ack map
+
+        :param ack_id: The ack id. Required.
+        :type ack_id: int
+        :return: The options.
+        :rtype: SendMessageErrorOptions or None
+        """
+        return self.ack_map.get(ack_id)
+
+    def clear(self) -> None:
+        """Clear ack map"""
+        for key, value in self.ack_map.items():
+            _LOGGER.debug("clear ack map with ack id: %s", key)
+            value.event.set()
+        self.ack_map.clear()
+
+
 class OpenClientError(AzureError):
     """Exception raised when fail to start the client"""
+
+
+class ReconnectError(AzureError):
+    """Exception raised when fail to reconnect"""
+
+
+class RecoverError(AzureError):
```

Comment:
Are these two exception types ever leaked publicly?

Reply:
see https://github.com/Azure/azure-sdk-for-python/pull/34764#discussion_r1568269497